contracts/staking/EpochSelector.sol

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./interfaces/IEpochSelector.sol";
+import "./ClusterSelector_wih_abi_encode.sol";
+
+/// @title Contract to select the top 5 clusters in an epoch
+contract EpochSelector is AccessControl, ClusterSelector, IEpochSelector {
+
+    using SafeERC20 for IERC20;
+
+    /// @notice Event emitted when Cluster is selected
+    /// @param epoch Number of Epoch
+    /// @param cluster Address of cluster
+    event ClusterSelected(uint256 indexed epoch, address indexed cluster);
+
+    /// @notice Event emited when the number of clusters to select is updated
+    /// @param newNumberOfClusters New number of clusters selected
+    event UpdateNumberOfClustersToSelect(uint256 newNumberOfClusters);
+
+    /// @notice Event emited when the reward is updated
+    /// @param newReward New Reward For selecting the tokens
+    event UpdateRewardForSelectingTheNodes(uint256 newReward);
+
+    /// @notice Event emited when the reward token is emitted
+    /// @param _newRewardToken Address of the new reward token
+    event UpdateRewardToken(address _newRewardToken);
+
+    /// @notice length of epoch
+    uint256 public constant epochLength = 4 hours;
+
+    /// @notice timestamp when the selector starts
+    uint256 public immutable startTime;
+
+    /// @notice Number of clusters selected in every epoch
+    uint256 public numberOfClustersToSelect;
+
+    /// @notice clusters selected during each epoch
+    mapping(uint256 => address[]) public clustersSelected;
```

Comment:
There should be a function which combines `selectCluster` and `updateMissingClusters` which is needed for verifying the tickets submitted instead of this.